tests/src/test/scala/org/apache/openwhisk/core/containerpool/test/ContainerProxyTests.scala

```diff
@@ -304,12 +307,15 @@ class ContainerProxyTests
       container.suspendCount shouldBe 0
       acker.calls should have size 2
 
-      // Add debugging information for the case of failure.
-      println(acker.calls.map(_._2.toJson.compactPrint))
-
       store.calls should have size 2
 
-      val initRunActivation = acker.calls(0)._2
+      // As the active acks are sent asynchronously, it is possible, that the activation with the init time is not the
+      // first one in the buffer.
+      val initializedActivations =
+        acker.calls.filter(_._2.annotations.get(WhiskActivation.initTimeAnnotation).isDefined)
+      initializedActivations should have size 1
+
+      val initRunActivation = initializedActivations.head._2
```

Comment:
As you already know that the size is 2 and you need both calls, could this be:

```scala
val (initRunActivation, runOnlyActivation) = {
  // false is sorted before true
  val sorted = acker.calls.sortBy(_._2.annotations.get(WhiskActivation.initTimeAnnotation).isEmpty)
  sorted(0)._2, sorted(1)._2
}
```

Reply:
That's a good point. 👍